server/app.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const path = require('path');
+const db = require('../db');
+
+const app = express();
+
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use('/books/:id', express.static(path.join(__dirname, '/../public')));
+
+// get and post routes to interact with database here
+
+// get all reviews for specific book id
+app.get('/books/:id/reviews', async (req, res) => {
+  const { id } = req.params;
+  try {
+    const reviews = await db.getReviews(id);
+    res.json(reviews);
+  } catch (err) {
+    res.json(err);
+  }
+});
+
+
+// get reviews for specific book w/ specific rating
+app.get('/books/:id/reviews/:rating', async (req, res) => {
+  const { id, rating } = req.params;
+  try {
+    const ratedReviews = await db.getRatedReviews(id, rating);
+    res.json(ratedReviews);
+  } catch (err) {
+    res.json(err);
+  }
+});
+
+// post review for specific book
+// app.post('/books/:id/reviews', (req, res) => {
+//   const { id } = req.params;
+//   const { review, rating } = req.body;
+
+//   db.postReview(id, review, rating).then(() => {
+//     db.getReviews(id);
+//   }).then((reviews) => {
+//     res.send(reviews);
+//   });
+// });
+
+// post review for specific book and get back all reviews w/ new review added
+app.post('/books/:id/reviews', async (req, res) => {
+  const { review, myRating } = req.body;
+  const { id } = req.params;
+
```

Comment:
I can't be sure but could it be the id you get from req.params is in string form? You could try console.log(typeof id) and check that it is in a number form. 

The way I understand it, if you want to put in a number into mysql, it has to be in number form when you add it to the [parameters] array to replace a question mark(?) during a db.query.

P.S. It would be helpful if you copy and pasted the exact error msg you are receiving.

Reply:
yeah, in my context note above I mentioned the id is in string form which is why I think it's not submitting. I wasn't sure why though because in my get request I was receiving it as a number. I tried adding removing the id from the params and inputting the params array and id separately but that didn't work either. 

 Below is error  message from postman:

{
    "code": "ER_PARSE_ERROR",
    "errno": 1064,
    "sqlMessage": "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'where book_id = '2'' at line 1",
    "sqlState": "42000",
    "index": 0,
    "sql": "insert into reviews (review, rating) values ('test review', '4') where book_id = '2'"
}

Reply:
if the Id is in string form you can just change it to a number using Number(id) function. I'm not sure why its in string form either if it was a number from your get request.